src/toil/batchSystems/kubernetes.py

```diff
@@ -677,44 +679,55 @@ def getUpdatedBatchJob(self, maxWait):
-                                    logger.debug("No job container statuses for job %s" % (pod.metadata.owner_references[0].name))
-                                    return UpdatedBatchJobInfo(jobID=int(pod.metadata.owner_references[0].name[len(self.jobPrefix):]), exitStatus=EXIT_STATUS_UNAVAILABLE_VALUE, wallTime=0, exitReason=None)
-
-                                # Get termination onformation from the pod
-                                termination = pod.status.container_statuses[0].state.terminated
-                                logger.info("REASON: %s Exit Code: %s", termination.reason, termination.exit_code)
-                                
-                                if termination.exit_code != 0:
-                                    # The pod failed. Dump information about it.
-                                    logger.debug('Failed pod information: %s', str(pod))
-                                    logger.warning('Log from failed pod: %s', self._getLogForPod(pod))
-                                jobID = int(pod.metadata.owner_references[0].name[len(self.jobPrefix):])
-                                terminated = pod.status.container_statuses[0].state.terminated
-                                runtime = slow_down((terminated.finished_at - terminated.started_at).total_seconds())
-                                result = UpdatedBatchJobInfo(jobID=jobID, exitStatus=terminated.exit_code, wallTime=runtime, exitReason=None)
-                                self._try_kubernetes(self._api('batch').delete_namespaced_job, 
-                                            pod.metadata.owner_references[0].name,
-                                            self.namespace,
-                                            propagation_policy='Foreground')
-
-                                self._waitForJobDeath(pod.metadata.owner_references[0].name)
-                                return result
-                            else:
-                                continue
+            for event in self._try_kubernetes_stream(self._api('batch').list_namespaced_job, self.namespace, 
+                                                        label_selector="toil_run={}".format(self.runID),
+                                                        timeout_seconds=maxWait):
+                # Grab the metadata data, ID and the list of conditions of the current job
+                jobObject = event['object']
+                jobID = int(jobObject.metadata.name[len(self.jobPrefix):])
+                jobObjectListConditions =jobObject.status.conditions
+                # Exit Reason defaults to 'Successfully Finsihed` unless said otherwise 
+                ExitReason = 1
+                ExitCode = 0
+
+                # Check if jobs exists by seeing if condition logs are pressent
+                if jobObjectListConditions is None or len(jobObjectListConditions):
```

Comment:
I think you need to check something about the event instead of assuming the job is done if it doesn't have any conditions. For one thing, we might be getting "added" or "deleted" events about new jobs being created or old jobs being successfully destroyed, not just "updated" events. For another, we can get an "updated" event when the job undergoes any change, not just when it finishes or fails.

We should ignore any events that aren't "updated" events, and we should ignore any events where the job isn't actually done running. I think you might be seeing jobs without `jobObject.status.conditions` set because the job is not really done yet.

You should check `jobObject.status.finished` and `jobObject.status.failed`. A job that is actually finished will have a nonzero value for one of those. Pending and running jobs will have 0 for both.